Added the SPDiagnosticsProvider resource

NikCharlebois opened the pull request:
Added a new resource to control and configure the Diagnostics Providers. #508 

- [ ] Change details added to Unreleased section of changelog.md?
- [ ] Added/updated documentation and descriptions in .schema.mof files where appropriate?
- [ ] Examples updated for both the single server and small farm templates in the examples folder?
- [ ] New/changed code adheres to [Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)?
- [ ] [Unit and Integration tests](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md) created/updated where possible?

ykuijs commented:
One question that is valid for all resources: In other resources, like SPAntivirusSettings, we didn't include an Ensure parameter simply because you cannot remove the config. In those resources we used $null values in the NullReturn to specify that the item didn't exist and checked for $null in the Test method.

You have chosen to use the Ensure and to throw an error when Absent is specified. Does this have an additional benefit? Did you have a specific reason for adding that parameter?

---

Reviewed 9 of 21 files at r1, 1 of 3 files at r2.
Review status: 10 of 21 files reviewed at latest revision, 16 unresolved discussions.

---

*[CHANGELOG.md, line 3 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31t5PddZLCkhbElf-b:-L31t5PejE9MWiNTBETF:bsom24g) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/CHANGELOG.md#L3)):*
> ```Markdown
> # Change log for SharePointDsc
> 
> ## 2.1
> ```

Please change the name of this section to Unreleased

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.psm1, line 101 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31tJR6txEdBbqUyChk:-L31tJR6txEdBbqUyChl:b-nofes8) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.psm1#L101)):*
> ```PowerShell
>     if($Ensure -eq "Absent")
>     {
>         throw "This ressource cannot remove Diagnostics Provider. Please use ensure equals Present."
> ```

ressource should be resource

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.psm1, line 115 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31vXZM-ReYyYv6xYDU:-L31vXZM-ReYyYv6xYDV:b-gzvdbl) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.psm1#L115)):*
> ```PowerShell
>         }
>         $diagnosticProvider.Retention = $params.Retention
>         $diagnosticProvider.MaxTotalSizeInBytes = $params.MaxTotalSizeInBytes
> ```

What happens if this parameter isn't specified? Wouldn't it be better to check if the parameter is specified before trying to set it?

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.schema.mof, line 8 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31tgdG_J_AswbB3dfW:-L31tgdG_J_AswbB3dfX:b-1gkqv7) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.schema.mof#L8)):*
> ```
>     [Write, Description("Sets the maximum retention size in bytes")] Uint64 MaxTotalSizeInBytes;
>     [Write, Description("True enables the Diagnostics Provider")] Boolean Enabled;
>     [Write, Description("Present to create this database, absent to ensure it does not exist"), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;
> ```

Shouldn't "Present to create this database" be something else? This doesn't do anything with databases, does it?

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/readme.md, line 4 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31tvonvOmGGt_0e96W:-L31tvonvOmGGt_0e96X:bysd7yf) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/readme.md#L4)):*
> ```Markdown
> 
> This resource is responsible for configuring the Diagnostics Provider within
> the local SharePoint farm.
> ```

Please also add a comment that mentions that Ensure=Absent is not supported.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1, line 191 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31uISGw76ftct5H8wm:-L31uISGw76ftct5H8wn:b-nofes8) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1#L191)):*
> ```PowerShell
>     if($Ensure -eq "Absent")
>     {
>         throw "This ressource cannot undo InfoPath Forms Service Configuration changes. `
> ```

ressource should be resource

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1, line 192 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31uOvOJQiQwJBDkyFE:-L31uOvOJQiQwJBDkyFF:b-og9s3g) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1#L192)):*
> ```PowerShell
>     {
>         throw "This ressource cannot undo InfoPath Forms Service Configuration changes. `
>         Please set Ensure to Present or ommit the resource"
> ```

ommit should be omit

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1, line 200 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31ueqTpbcORlo-t3WF:-L31ueqTpbcORlo-t3WG:b-pkpxy2) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1#L200)):*
> ```PowerShell
>         $params = $args[0]
>         $config = Get-SPInfoPathFormsService
>         $config.AllowUserFormBrowserEnabling = $params.AllowUserFormBrowserEnabling
> ```

By specifying a default parameter value and always setting it, you cannot change just one setting. All others are then set to default.

Wouldn't it be better to only set a parameter when the parameter is actually provided? Similar to what we do in SPAntivirusSettings.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/readme.md, line 4 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31vvDyf6b0AbweBkvK:-L31vvDyf6b0AbweBkvL:bysd7yf) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/readme.md#L4)):*
> ```Markdown
> 
> This resource is responsible for configuring the InfoPath Forms service within
> the local SharePoint farm.
> ```

Please also add a comment that mentions that Ensure=Absent is not supported.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 111 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31w2EI97BuTbQ8nI-T:-L31w2EI97BuTbQ8nI-U:b-nofes8) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L111)):*
> ```PowerShell
>     if($Ensure -eq "Absent")
>     {
>         throw "This ressource cannot undo Security Token Service Configuration changes. `
> ```

ressource should be resource

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 112 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31w42XebahLoV-XOoG:-L31w42XebahLoV-XOoH:b-og9s3g) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L112)):*
> ```PowerShell
>     {
>         throw "This ressource cannot undo Security Token Service Configuration changes. `
>         Please set Ensure to Present or ommit the resource"
> ```

ommit should be omit

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 121 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31wAHKvoUk97HM40NM:-L31wAHLJnZrV0A3P66v:b-pkpxy2) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L121)):*
> ```PowerShell
>         $config = Get-SPSecurityTokenServiceConfig
>         $config.Name = $params.Name
>         $config.NameIdentifier = $params.NameIdentifier
> ```

By specifying a default parameter value and always setting it, you cannot change just one setting. All others are then set to default.

Wouldn't it be better to only set a parameter when the parameter is actually provided? Similar to what we do in SPAntivirusSettings.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.schema.mof, line 9 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L320ddgVl6oeSw2E-U3:-L320ddgVl6oeSw2E-U4:b-odbhgc) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.schema.mof#L9)):*
> ```
>     [Write, Description("True set the security token service to allow OAuth over HTTP")] Boolean AllowOAuthOverHttp;
>     [Write, Description("True set the security token service to allow metadata exchange over HTTP")] Boolean AllowMetadataOverHttp;
>     [Write, Description("Present ensures service app exists, absent ensures it is removed"), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;
> ```

Description is talking about service app, which is not correct.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/readme.md, line 4 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L320rRHcb-KfCjIrJDw:-L320rRHcb-KfCjIrJDx:bysd7yf) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/readme.md#L4)):*
> ```Markdown
> 
> This resource is responsible for configuring the Security Token Service within
> the local SharePoint farm.
> ```

Please also add a comment that mentions that Ensure=Absent is not supported.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPUserProfileSyncConnection/MSFT_SPUserProfileSyncConnection.psm1, line 331 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L323T497YRzzKHLaxGM:-L323T497YRzzKHLaxGN:b-hrgtbj) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPUserProfileSyncConnection/MSFT_SPUserProfileSyncConnection.psm1#L331)):*
> ```PowerShell
>                         Add-SPProfileSyncConnection -ProfileServiceApplication $ups -ConnectionForestName $params.Forest -ConnectionDomain $userDomain `
>                             -ConnectionUserName $userName -ConnectionPassword $params.ConnectionCredentials.Password -ConnectionUseSSL $params.UseSSL `
>                             -ConnectionSynchronizationOU $ou
> ```

I do not see any changes for removing OU's. What happens if I change the OU list? Are OU's that are removed from the parameter also removed from the config in SP?

---

*[Modules/SharePointDsc/Examples/Resources/SPInfoPathFormsServiceConfig/1-Configure.ps1, line 3 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L323jNV6-1eYCR5R4Fq:-L323jNV6-1eYCR5R4Fr:bqtazyh) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/Examples/Resources/SPInfoPathFormsServiceConfig/1-Configure.ps1#L3)):*
> ```PowerShell
> <#
> .EXAMPLE
>     This example shows how to deploy Access Services 2010 to the local SharePoint farm.
> ```

Please update this text

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-:-L324JZHIwhRyR-qlBrw:b-xvcdji)*
<!-- Sent from Reviewable.io -->

NikCharlebois commented:
The reason why I decided to use Ensure, is because these resource don't have a primary key, because "Name" cannot be used. I was looking at other resources that don't have a Primary key, and were using Ensure as an approach instead. Not sure how we want to handle this moving forward.

---

Review status: 10 of 21 files reviewed at latest revision, 16 unresolved discussions.

---

*[CHANGELOG.md, line 3 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31t5PddZLCkhbElf-b:-L334s3ASsTwx1tDnrnB:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/CHANGELOG.md#L3)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Please change the name of this section to Unreleased
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.psm1, line 101 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31tJR6txEdBbqUyChk:-L334rMCDkpF-4dnbstm:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.psm1#L101)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

ressource should be resource
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.psm1, line 115 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31vXZM-ReYyYv6xYDU:-L335H0m_p12AG8nblx4:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.psm1#L115)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

What happens if this parameter isn't specified? Wouldn't it be better to check if the parameter is specified before trying to set it?
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.schema.mof, line 8 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31tgdG_J_AswbB3dfW:-L335S590wktqmKLf-PS:b-5540md) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/MSFT_SPDiagnosticsProvider.schema.mof#L8)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Shouldn't "Present to create this database" be something else? This doesn't do anything with databases, does it?
</blockquote></details>

Yep it should have. That was an oversight.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/readme.md, line 4 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31tvonvOmGGt_0e96W:-L335ff5M4SgVCQ1jtsx:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPDiagnosticsProvider/readme.md#L4)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Please also add a comment that mentions that Ensure=Absent is not supported.
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1, line 191 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31uISGw76ftct5H8wm:-L335hrpp0BecTv1zXzL:b-14whu7) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1#L191)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

ressource should be resource
</blockquote></details>

This is my French background showing :) Done

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1, line 192 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31uOvOJQiQwJBDkyFE:-L335vB4LtMUScYohgGN:b-xudied) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1#L192)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

ommit should be omit
</blockquote></details>

I will be looking around at other resources to fix this issue as well, I have seen this in multiple other occasions.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1, line 200 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31ueqTpbcORlo-t3WF:-L3388stJ9LwjLIHbcD2:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1#L200)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

By specifying a default parameter value and always setting it, you cannot change just one setting. All others are then set to default.

Wouldn't it be better to only set a parameter when the parameter is actually provided? Similar to what we do in SPAntivirusSettings.
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/readme.md, line 4 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31vvDyf6b0AbweBkvK:-L338OEMg2Dmcc5SmcHR:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/readme.md#L4)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Please also add a comment that mentions that Ensure=Absent is not supported.
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 111 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31w2EI97BuTbQ8nI-T:-L338kXzhaTEaNkaJMns:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L111)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

ressource should be resource
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 112 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31w42XebahLoV-XOoG:-L338nmfqDjSgeeTATTr:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L112)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

ommit should be omit
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 121 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31wAHKvoUk97HM40NM:-L338oojBjYmisiKbDGR:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L121)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

By specifying a default parameter value and always setting it, you cannot change just one setting. All others are then set to default.

Wouldn't it be better to only set a parameter when the parameter is actually provided? Similar to what we do in SPAntivirusSettings.
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.schema.mof, line 9 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L320ddgVl6oeSw2E-U3:-L338rXCa8CAeCe3hNQE:b-fs2m0o) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.schema.mof#L9)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Description is talking about service app, which is not correct.
</blockquote></details>

Yes, correct. It is dealing with configuration. Made the changes

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/readme.md, line 4 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L320rRHcb-KfCjIrJDw:-L3399gUsmwV4mHfoKnV:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/readme.md#L4)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Please also add a comment that mentions that Ensure=Absent is not supported.
</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPUserProfileSyncConnection/MSFT_SPUserProfileSyncConnection.psm1, line 331 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L323T497YRzzKHLaxGM:-L33A3SB6Zwh7s4f4nd6:b-w9p9n7) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPUserProfileSyncConnection/MSFT_SPUserProfileSyncConnection.psm1#L331)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

I do not see any changes for removing OU's. What happens if I change the OU list? Are OU's that are removed from the parameter also removed from the config in SP?
</blockquote></details>

Added a Remove-SPProfileSyncConnection block

---

*[Modules/SharePointDsc/Examples/Resources/SPInfoPathFormsServiceConfig/1-Configure.ps1, line 3 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L323jNV6-1eYCR5R4Fq:-L33CKK0GI9fpOYVe_9l:b-li4q9c) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/Examples/Resources/SPInfoPathFormsServiceConfig/1-Configure.ps1#L3)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Please update this text
</blockquote></details>

Wow, totally missed that one. Thanks

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-:-L333gB8y9lBgfwOvsBk:b-a6qrsb)*
<!-- Sent from Reviewable.io -->

ykuijs commented:
Added one more comment in the Changelog and two follow-up questions for other comments.

The SPDiagnosticsProvider and SPSecurityTokenServiceConfig resources do have Name parameters which can be configured as Key, so I prefer them to be used as Key and the Ensure removed. You are right for the InfoPath resource.

---

Reviewed 1 of 21 files at r1, 11 of 14 files at r3.
Review status: 18 of 21 files reviewed at latest revision, 3 unresolved discussions.

---

*[CHANGELOG.md, line 5 at r3](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L349OOsODXimeGyqoRZ:-L349OOsODXimeGyqoR_:b-rq9d17) ([raw file](https://github.com/powershell/sharepointdsc/blob/81f0728bef274214bf564ce5e95ce9ee6d9354e1/CHANGELOG.md#L5)):*
> ```Markdown
> ## Unreleased
> 
> * Added the SPDiagnosticsProvider resource
> ```

Since the v2.0 release I grouped the changes per resource. Can you change your changes into:
* SPInfoPathFormServiceConfig
  * New resource

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1, line 200 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31ueqTpbcORlo-t3WF:-L34AC_G59tIPJULmiYG:b-56pmt8) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1#L200)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Doesn't the check always return true, since the parameters have default values configured?

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 121 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31wAHKvoUk97HM40NM:-L34AdpyWdy5AQUp39Pz:b-56pmt8) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L121)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Doesn't the check always return true, since the parameters have default values configured?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-:-L34C6QeAK10pmtf55Me:b-5peblf)*
<!-- Sent from Reviewable.io -->

NikCharlebois commented:
Review status: 18 of 21 files reviewed at latest revision, 3 unresolved discussions.

---

*[CHANGELOG.md, line 5 at r3](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L349OOsODXimeGyqoRZ:-L38EoL0M6DS43Gd57VO:b-896fix) ([raw file](https://github.com/powershell/sharepointdsc/blob/81f0728bef274214bf564ce5e95ce9ee6d9354e1/CHANGELOG.md#L5)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Since the v2.0 release I grouped the changes per resource. Can you change your changes into:
* SPInfoPathFormServiceConfig
  * New resource

</blockquote></details>

Done.

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1, line 200 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31ueqTpbcORlo-t3WF:-L38FvPvMZF6gKpkrvu6:b-z2ljpo) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPInfoPathFormsServiceConfig/MSFT_SPInfoPathFormsServiceConfig.psm1#L200)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Doesn't the check always return true, since the parameters have default values configured?
</blockquote></details>

You are absolutely right. I removed the default values in all 3 new resources. I didn't think about that case, Great catch

---

*[Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 121 at r2](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-L31wAHKvoUk97HM40NM:-L38G28uTcLPXWNyQc05:bb6un1j) ([raw file](https://github.com/powershell/sharepointdsc/blob/181214c060f6aaa84d0bc41cca00bfe209bb09d1/Modules/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L121)):*
<details><summary><i>Previously, ykuijs (Yorick Kuijs) wrote…</i></summary><blockquote>

Doesn't the check always return true, since the parameters have default values configured?
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/sharepointdsc/729)*
<!-- Sent from Reviewable.io -->

NikCharlebois commented:
All set for final review

ykuijs commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 14 files at r3, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/sharepointdsc/729#-:-L39nV-r1ixsf21EiQtG:bnfp4nl)*
<!-- Sent from Reviewable.io -->
